### ISSUE TITLE
More OpenAI configurability

### ISSUE THREAD

**Issue by KTibow** [zed-industries/zed#22030]
### Check for existing issues

- [X] Completed

### Describe the feature

tl;dr Allow disabling `gpt-4o` and the lot, using separate APIs, and perhaps reordering/disabling models

For context, if you reconfigure your OpenAI base URL, the models get really pushed down. You also can't have two different OpenAI base URLs.
<img src="https://github.com/user-attachments/assets/15e1ad3f-ead0-4fb5-b2e4-30cdeef0111a" height="400px" />

Continue to the mockup section for some of my ideas.

### Environment

```
Zed: v0.165.4 (Zed)
OS: Linux Wayland fedora 40
Memory: 15.2 GiB
Architecture: x86_64
GPU: Intel(R) Iris(R) Plus Graphics (ICL GT1.5) || Intel open-source Mesa driver || Mesa 24.1.7
```

### If applicable, add mockups / screenshots to help present your vision of the feature

The main thing I'd like is a way to disable the default models. This might look like
```json
"use_default_models": false,
```

The second thing I'd like is to allow using per-model URLs. This might look like
```json
      "available_models": [
        {
          "name": "llama-3.3-70b-specdec",
          "display_name": "Llama",
          "max_tokens": 6000,
          "api_url": "https://api.groq.com/openai/v1/"
        },
        {
          "name": "gpt-4o",
          "display_name": "GPT-4o",
          "max_tokens": 16000,
          "api_url": "https://api.openai.com/v1/"
        }
      ]
```

I'm not quite sure what the third thing would look like. It could be a list of models that gets saved to configuration:
```
    "model_order": ["Claude 3.5 Sonnet", "Llama", "GPT-4o"]
```
Or it could be more graphical, like extending the current "Configure your assistant" page so each assistant has drag handles and `x` buttons.

**Comment by notpeter** [comment]
Being able to configure multiple endpoints/apikeys for the same underlying "provider" is a duplicate of:
- https://github.com/zed-industries/zed/issues/15732

As the number of models/providers grows the single monolithic list of models is definitely becoming harder to use.  There have been some mocks of alternatives done internally (multiple dropdowns; folding by provider; pinning specific models, hiding models, etc) but we haven't landed on anything specific yet.  I'm going go ahead and migrate this to a discussion in case folks have further ideas.

Thanks for reporting.
